### ISSUE TITLE
xbps-src does not build depends of subpackages

### ISSUE THREAD

**Issue by Chocimier** [void-linux/void-packages#20508]
This can lead to indexing package without its runtime dependencies being available (broken, misspelled, not yet build).
So depends of subpackages should be build, same as depends of main package.

On the other hand, building them can introduce cycles.

**Comment by CameronNemo** [comment]
This is particularly troublesome because subpackages more often than not depend on the main package.

**Comment by xtraeme** [comment]
The correct way is adding all build dependencies in the main pkg.

**Comment by Duncaen** [comment]
Its about runtime dependencies, the assumption that all packages in the index are installable (ignoring ongoing shlib bumps) breaks because sub packages can have runtime dependencies that have not been built at the time xbps-src adds them to the index.

**Comment by xtraeme** [comment]
@Duncaen I'm aware. That's why the main pkg should have all deps. Maybe via `hostmakedepends`

**Comment by Duncaen** [comment]
I think it would probably more intuitive to just include depends on sub packages when checking build dependencies.
Using `hostmakedepends` would be for the wrong architecture, `makedepends` would work but as with `hostmakedepends` this adds extra useless installation time.

**Comment by xtraeme** [comment]
Another idea I had a while ago is, make `xbps-rindex -a` error out if it cannot resolve all dependencies... this would fix 99% of the issues.

**Comment by xtraeme** [comment]
@Duncaen sounds good, but as @Chocimier mentions above it will introduce lots of cycles.

**Comment by Chocimier** [comment]
Depending on subpackage (like -data) from main package is already supported, so this won't be a problem.

What about treating depends of subpackages exactly like depends of main package, ie. build if missing but not install?

Checked for cycles. There is one connected component of 94 packages. Looks like breakable. Here is [tool I used](https://gist.github.com/Chocimier/de76441493ec7775c201dac0bb03ced5).

ModemManager NetworkManager SDL2 attica avahi bluez cargo colord cups dav1d doxygen elogind extra-cmake-modules ffmpeg flac fontforge gcr ghostscript git gnupg2 gpgme graphviz gst-plugins-base1 gstreamer1 gtk+ gtk+3 jack karchive kauth kbookmarks kcodecs kcompletion kconfig kconfigwidgets kcoreaddons kcrash kdbusaddons kded kdelibs4support kdesignerplugin kdoctools kemoticons kglobalaccel kguiaddons ki18n kiconthemes kinit kio kitemmodels kitemviews kjobwidgets knotifications kparts kplotting kservice ksolid ktextwidgets kunitconversion kwallet kwidgetsaddons kwindowsystem kxmlgui lame libblockdev libbs2b libdbusmenu-qt5 liberation-fonts-ttf libgit2 libinput librsvg libsamplerate libsndfile libwacom nasm net-tools networkmanager-qt5 ocl-icd phonon-qt5 pinentry polkit polkit-qt5 pulseaudio qt5 ruby sane sbc sonnet subversion udisks2 upower volume_key x264 x265 xvidcore
